tools/rosbag_storage/src/bag.cpp

```diff
@@ -173,6 +173,21 @@ void Bag::close() {
     chunks_.clear();
     connection_indexes_.clear();
     curr_chunk_connection_indexes_.clear();
+
+    mode_ = bagmode::Write;
+    version_ = 0;
+    compression_ = compression::Uncompressed;
+    chunk_threshold_ = 768 * 1024;  // 768KB chunks
+    bag_revision_ = 0;
+    file_size_ = 0;
+    file_header_pos_ = 0;
+    index_data_pos_ = 0;
+    connection_count_ = 0;
+    chunk_count_ = 0;
+    chunk_open_ = false;
+    curr_chunk_data_pos_ = 0;
+    current_buffer_ = 0;
+    decompressed_chunk_ = 0;
```

Comment:
This should be easier now that #1189 has been merged. Please update the patch to call the new `init` method instead.